kobo/apps/long_running_migrations/jobs/0007_backfill_attachment_model.py

```diff
@@ -1,61 +1,74 @@
 # Generated by Django 4.2.15 on 2025-03-12 14:37
-from more_itertools import chunked
+from django.db.models import Q
 
 from kobo.apps.openrosa.apps.logger.models.attachment import (
     Attachment,
     AttachmentDeleteStatus,
 )
 from kpi.fields.kpi_uid import KpiUidField
 
+CHUNK_SIZE = 2000
+
 
 def run():
     """
     Backfills xform_id, user_id, uid, date_created, and date_modified for Attachment
     model.
     """
-    CHUNK_SIZE = 2000
+    stop = False
+    attachment_pk = 0
 
-    attachments = (
-        Attachment.all_objects.only(
-            'instance_id',
+    while not stop:
+        if attachments := get_queryset(attachment_pk):
+            for attachment in attachments:
+                instance = attachment.instance
+
+                attachment.xform_id =  instance.xform_id
+                attachment.user_id = instance.xform.user_id
+
+                if not attachment.uid:
+                    attachment.uid = KpiUidField.generate_unique_id('att')
+
+                if not attachment.date_created:
+                    attachment.date_created = instance.date_created
+
+                if not attachment.date_modified:
+                    attachment.date_modified = attachment.date_created
+
+                if attachment.deleted_at and not attachment.delete_status:
+                    attachment.date_modified = attachment.deleted_at
+                    attachment.delete_status = AttachmentDeleteStatus.SOFT_DELETED
+
+            Attachment.all_objects.bulk_update(
+                attachments,
+                fields=[
+                    'xform_id',
+                    'user_id',
+                    'uid',
+                    'date_created',
+                    'date_modified',
+                    'delete_status',
+                ],
+            )
+            attachment_pk = attachment.pk
```

Comment:
Even it is going to be set multiple time nothing, let's move `attachment_pk = attachment.pk` inside the loop , right below `attachment.user_id = instance.xform.user_id`